cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4CareGapsBundleBuilder.java

```diff
@@ -0,0 +1,322 @@
+
+        List<ParametersParameterComponent> paramResults = new ArrayList<>();
+
+        for (String subject : subjects) {
+            // Measure Reports
+            Bundle result = r4MultiMeasureService.evaluate(
+                    measureIds,
+                    null,
+                    null,
+                    periodStart.getValueAsString(),
+                    periodEnd.getValueAsString(),
+                    MeasureEvalType.SUBJECT.toCode(),
+                    subject,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    reporter);
+
+            // Patient
+            var patient = repository.read(Patient.class, new IdType(subject));
+
+            // finalize patient Bundle results
+            var bundle = makePatientBundle(result, statuses, patient);
+
+            // add parameter with results
+            if (bundle != null && bundle.hasEntry()) {
+                paramResults.add(initializePatientParameter(patient).setResource(bundle));
+            }
+        }
+        return paramResults;
+    }
+
+    public Bundle makePatientBundle(Bundle bundle, List<String> statuses, Patient patient) {
+        Map<String, Resource> evalPlusSDE = new HashMap<>();
+        List<DetectedIssue> detectedIssues = new ArrayList<>();
+        List<MeasureReport> measureReports = new ArrayList<>();
+        var gapEvaluator = new R4CareGapStatusEvaluator();
+        Composition composition = getComposition(patient);
+        // get Evaluation Bundle Results
+        for (BundleEntryComponent entry : bundle.getEntry()) {
+            MeasureReport mr = (MeasureReport) entry.getResource();
+            addProfile(mr);
+            addResourceId(mr);
+            Measure measure = r4MeasureServiceUtils.resolveByUrl(mr.getMeasure());
+            // Applicable Reports per Gap-Status
+            var gapStatus = gapEvaluator.getGapStatus(measure, mr);
+            boolean keepResult = statuses.contains(gapStatus.toString());
+            if (keepResult) {
+                // add Report to final Care-gap report
+                measureReports.add(mr);
+                // Issue Detected for Report
+                DetectedIssue detectedIssue = getDetectedIssue(patient, mr, gapStatus);
+                detectedIssues.add(getDetectedIssue(patient, mr, gapStatus));
+                composition.addSection(getSection(measure, mr, detectedIssue, gapStatus));
+                // Track evaluated Resources
+                populateEvaluatedResources(mr, evalPlusSDE);
+                populateSDEResources(mr, evalPlusSDE);
+            }
+        }
+
+        if (!measureReports.isEmpty()) {
+            // only add if a DetectedIssue is found and has MeasureReports
+            return addBundleEntries(serverBase, composition, detectedIssues, measureReports, evalPlusSDE);
+        } else {
+            return null;
```

Comment:
Just curious:  why not return an empty Bundle instead of null?  If you still plan on returning null, please mark this method as @Nullable

Reply:
that is the designed behavior, if report is not applicable to the status requested for the patient then the report comes back empty